backend/supabase/functions/complete-bonus-game/index.ts

```diff
@@ -0,0 +1,572 @@
+import { serve } from 'https://deno.land/std@0.168.0/http/server.ts'
+import { ethers } from 'https://esm.sh/ethers@6'
+import { handleCorsPreFlight } from '../_shared/cors.ts'
+import { successResponse, errorResponse } from '../_shared/response.ts'
+import { validateRequired, compareAddresses } from '../_shared/validation.ts'
+import { initSupabaseClient } from '../_shared/supabase.ts'
+import {
+  HOOT_BONUS_QUIZ_MANAGER_ABI,
+  ERC20_ABI,
+  ZERO_ADDRESS,
+  GAME_STATUS,
+  QUIZ_STATUS
+} from '../_shared/constants.ts'
+import type { CompleteGameRequest, GameSession, PlayerSession } from '../_shared/types.ts'
+
+async function fetchGameSession(supabase: ReturnType<typeof initSupabaseClient>, gameSessionId: string): Promise<GameSession | null> {
```

Comment:
@cursoragent why are you using this if you have an helper in _shared/database folder